rare/game_launch_helper/lgd_helper.py

```diff
@@ -0,0 +1,152 @@
+    if platform.system() == "Windows":
+        launch_args.executable = origin_uri
+        launch_args.args = []
+        # only set it here true, because on linux it is a launch command like every other game
+        launch_args.is_origin_game = True
+        return launch_args
+
+    command = core.get_app_launch_command(app_name)
+    if not os.path.exists(command[0]) and shutil.which(command[0]) is None:
+        return launch_args
+    command.append(origin_uri)
+
+    launch_args.executable = command[0]
+    launch_args.args = command[1:]
+    return launch_args
+
+
+def get_game_params(core: LegendaryCore, igame: InstalledGame, offline: bool,
+                    skip_update_check: bool, launch_args: LaunchReturnValue) -> LaunchReturnValue:
+    if not offline:  # skip for update
+        if not skip_update_check and not core.is_noupdate_game(igame.app_name):
+            # check updates
+            try:
+                latest = core.get_asset(
+                    igame.app_name, igame.platform, update=False
+                )
+            except ValueError:
+                logger.error("Metadata doesn't exist")
+                return launch_args
+            else:
+                if latest.build_version != igame.version:
+                    return launch_args
+    params: LaunchParameters = core.get_launch_parameters(
+        app_name=igame.app_name, offline=offline
+    )
+
+    full_params = list()
+
+    if os.environ.get("container") == "flatpak":
+        full_params.extend(["flatpak-spawn", "--host"])
+
+    full_params.extend(params.launch_command)
+    full_params.append(
+        os.path.join(params.game_directory, params.game_executable)
+    )
+    full_params.extend(params.game_parameters)
+    full_params.extend(params.egl_parameters)
+    full_params.extend(params.user_parameters)
+
+    launch_args.executable = full_params[0]
+    launch_args.args = full_params[1:]
+
+    return launch_args
+
+
+def get_launch_args(core: LegendaryCore, args: LaunchArgs = None) -> LaunchReturnValue:
+    game = core.get_game(args.app_name)
+    igame = core.get_installed_game(args.app_name)
+
+    resp = LaunchReturnValue()
+
+    if not game:
+        return resp
+
+    if game.third_party_store == "Origin":
+        args.offline = False
+    else:
+        if not igame:
+            logger.error("Game is not installed or has unsupported format")
+            return resp
+
+        if game.is_dlc:
+            logger.error("Game is dlc")
+            return resp
+        if not os.path.exists(igame.install_path):
+            logger.error("Game path does not exist")
+            return resp
+
+    if game.third_party_store == "Origin":
+        resp = get_origin_params(core, args.app_name, args.offline, resp)
+    else:
+        resp = get_game_params(core, igame, args.offline, args.skip_version_check, resp)
+
+    env = core.get_app_environment(args.app_name, wine_pfx=args.wine_prefix)
+    environment = QProcessEnvironment()
```

Comment:
`QProcessEnvironment()` creates a clean environment. `core.get_app_environment()` doesn't contain the system's environment, anything the user might have specified outside of Legendary config won't be honored.

Reply:
On another note, the return of `core.get_launch_parameters` also contains the environment, can't this process be refactored to not request the same data multiple times?

Reply:
get_app_environment and get_launch_parameters?

Reply:
I meant `get_origin_params`, `get_game_params` and `get_launch_args`